pyface/ui/qt4/python_shell.py

```diff
@@ -118,9 +118,11 @@ def _add_event_listeners(self):
     def _remove_event_listeners(self):
         if self.control is not None:
             # Disconnect signals for events.
-            self.control.executed.connect(self._on_command_executed)
+            self.control.executed.disconnect(self._on_command_executed)
             self._event_filter.signal.disconnect(self._on_obj_drop)
 
+            self.control.disconnect_event_listeners()
+
```

Comment:
Nit:  Rename `disconnect_event_listeners` to `_remove_event_listeners`?

Reply:
I also moved the methods closer to `__init__` and removed the "public interface" headers